runtime/ftplugin/javascript.vim

```diff
@@ -26,13 +27,63 @@ setlocal comments=sO:*\ -,mO:*\ \ ,exO:*/,s1:/*,mb:*,ex:*/,://
 
 setlocal commentstring=//%s
 
-" Change the :browse e filter to primarily show Java-related files.
+" Change the :browse e filter to primarily show JavaScript-related files.
 if has("gui_win32")
-    let  b:browsefilter="Javascript Files (*.js)\t*.js\n" .
-		\	"All Files (*.*)\t*.*\n"
+    let  b:browsefilter="Javascript Files (*.js, *.jsx, *.es, *.es6, *.cjs, *.mjs, *.jsm, *.vue, *.json)\t*.js;*.jsx;*.es;*.es6;*.cjs;*.mjs;*.jsm;*.vue;*.json\n" .
+                \ "All Files (*.*)\t*.*\n"
 endif
-       
-let b:undo_ftplugin = "setl fo< ofu< com< cms<" 
+
+" The following suffixes should be implied when resolving filenames
+setlocal suffixesadd+=.js,.jsx,.es,.es6,.cjs,.mjs,.jsm,.vue,.json
+
+" The following suffixes should have low priority
+"   .snap    jest snapshot
+setlocal suffixes+=.snap
+
+" Prepend node_modules/.bin to $PATH if applicable
+" Allows calling executables installed locally via npm/yarn
+let s:bin_dir = finddir('node_modules/.bin', '.;')->fnamemodify(':p')
+if len(s:bin_dir) && $PATH !~ s:bin_dir
+    let $PATH = s:bin_dir .. ':' .. $PATH
+endif
+unlet s:bin_dir
+
+" Set 'path' to a minimal, non-greedy, value
+" User is expected to augment it with contextually-relevant paths
+setlocal path-=node_modules
+setlocal path-=/usr/include
+setlocal path-=**
+
```

Comment:
> Where were they added?

Somewhere in the user's `vimrc`.

> Why remove them?

Because they make everything slower. `**` is a very bad idea that perverts the very idea of `&path`, `node_modules` is black hole, and `/usr/include` is irrelevant.

Reply:
- `**` is too greedy and completely defeats the purpose of `&path` which is to explicitly list interesting directories from which to look for files. `dist/` or `coverage/` are totally irrelevant so they shouldn't be searched.
- And the effect of `**` is aggravated by `node_modules/` which is, again, a black hole for Vim's search algorithm. That's as "JavaScript" as it gets, IMO.

I could settle for only removing `/usr/include` and letting the user shoot themselves in the foot if they so choose.

Reply:
Unfortunately, users put `**` in their `&path` because they were told to do it by other misinformed users. `set path+=**` is widely used despite it breaking lots of commands. You would know it if you spent as much time as me providing support. This is only an attempt to alleviate all too common problems.

Anyway, as I said I will just remove it.

Reply:
DONE in commit 850c35f